internal/osbuild/gcp_guest_agent_conf_stage.go

```diff
@@ -0,0 +1,118 @@
+package osbuild
+
+import (
+	"fmt"
+)
+
+type GcpGuestAgentConfigScopeValue string
+
+const (
+	GcpGuestAgentConfigScopeDistro   GcpGuestAgentConfigScopeValue = "distro"
+	GcpGuestAgentConfigScopeInstance GcpGuestAgentConfigScopeValue = "instance"
+)
+
+type GcpGuestAgentConfigAccounts struct {
+	GcpGuestAgentConfigDeprovisionRemove *bool    `json:"deprovision_remove,omitempty"`
+	GcpGuestAgentConfigGroups            []string `json:"groups,omitempty"`
+	GcpGuestAgentConfigUseraddCmd        string   `json:"useradd_cmd,omitempty"`
+	GcpGuestAgentConfigUserdelCmd        string   `json:"userdel_cmd,omitempty"`
+	GcpGuestAgentConfigUsermodCmd        string   `json:"usermod_cmd,omitempty"`
+	GcpGuestAgentConfigGpasswdAddCmd     string   `json:"gpasswd_add_cmd,omitempty"`
+	GcpGuestAgentConfigGpasswdRemoveCmd  string   `json:"gpasswd_remove_cmd,omitempty"`
+	GcpGuestAgentConfigGroupaddCmd       string   `json:"groupadd_cmd,omitempty"`
```

Comment:
Sorry if my previous comments were confusing. I think that adding the stage name into the exported structures names makes the code more readable, but there is IMO no point in prefixing each structure member name with it as well. So `DeprovisionRemove` should be preferred over `GcpGuestAgentConfigDeprovisionRemove`. My point was mainly that if the structure itself would not have the prefix, it would not be clear to which stage it is related. This is not the case with members of the structure.

Reply:
Oh no that's on me, I took this way to literal.
Fixed it up.